build_image.sh

```diff
@@ -87,11 +87,25 @@ generate_onie_installer_image()
           $ONIE_INSTALLER_PAYLOAD
 }
 
+generate_device_list()
+{
+    # Generate asic-specific device list
+    local platforms_asic=$1
+    for d in `find -L ./device  -maxdepth 2 -mindepth 2 -type d`; do
+        if [ -f $d/platform_asic ] && grep -Fxq "$CONFIGURED_PLATFORM" $d/platform_asic; then
+            echo "${d##*/}" >> "$platforms_asic";
```

Comment:
>echo "${d##*/}" >> "$platforms_asic"; [](http://example.com/codeflow?start=12&length=37)

You need to clean the file at the begining in case the file is not empty. No need to `rm` before this function call.
If there is no platforms using CONFIGURED_PLATFORM asic, this function should generated an empty file.

Reply:
In case of general build, sonic installer verification would bypass because $platforms_asic file is not there. But if it is a empty file, it will fail because no platform is in the file. I guess general build means we could install it to any asic.
CONFIGURED_PLATFORM is general by default.
Check out here. [general](https://github.com/Azure/sonic-buildimage/blob/master/onie-image.conf#L11)